llvm/lib/Transforms/Utils/SimplifyCFG.cpp

```diff
@@ -2335,6 +2345,10 @@ static bool sinkCommonCodeFromPredecessors(BasicBlock *BB,
               return InstructionsToSink.contains(V);
             })) {
           ++NumPHIInsts;
+          // Do not separate a load/store from the gep producing the address.
```

Comment:
If I'm following correctly, this is preventing the fold in cases where the GEP would likely fold into the load/store instruction, and the GEP itself won't sink?  That probably makes sense, given the GEP is very likely to fold into the load/store, and it's hard to analyze PHIs.

Maybe expand the comment here a bit.